### PR TITLE
Check that the document can be changed for a file rename

### LINKED ISSUE #38525: Inline rename with file rename fails in a shared project
**Version Used**: 

**Steps to Reproduce**:

1. Create a C# shared project
2. Add a new class named Class1 and file Class1.cs
3. Try to rename the class to something else. Make sure that "Rename file" is checked, otherwise the issue does not repro.

**Expected Behavior**:
Rename succeeds

**Actual Behavior**:
Rename fails with the following exception:

```
System.NotSupportedException: Changing document properties is not supported   at Microsoft.CodeAnalysis.Workspace.CheckAllowedProjectChanges(ProjectChanges projectChanges)
   at Microsoft.CodeAnalysis.Workspace.CheckAllowedSolutionChanges(SolutionChanges solutionChanges)
   at Microsoft.CodeAnalysis.Workspace.TryApplyChanges(Solution newSolution, IProgressTracker progressTracker)
   at Microsoft.CodeAnalysis.Workspace.TryApplyChanges(Solution newSolution)
   at Microsoft.CodeAnalysis.Editor.Implementation.InlineRename.InlineRenameSession.ApplyRename(Solution newSolution, IWaitContext waitContext)
   at Microsoft.CodeAnalysis.Editor.Implementation.InlineRename.InlineRenameSession.CommitCore(IWaitContext waitContext, Boolean previewChanges)
   at Microsoft.CodeAnalysis.Editor.Implementation.InlineRename.InlineRenameSession.<>c__DisplayClass76_0.<Commit>b__0(IWaitContext waitContext)
   at Microsoft.VisualStudio.LanguageServices.Implementation.Utilities.VisualStudioWaitIndicator.Wait(String title, String message, Boolean allowCancel, Boolean showProgress, Action`1 action)
   at Microsoft.CodeAnalysis.Editor.Implementation.InlineRename.InlineRenameSession.Commit(Boolean previewChanges)
   at Microsoft.CodeAnalysis.Editor.Implementation.InlineRename.RenameCommandHandler.ExecuteCommand(ReturnKeyCommandArgs args, CommandExecutionContext context)
   at Microsoft.VisualStudio.Commanding.CommandHandlerExtensions.ExecuteCommand[T](ICommandHandler commandHandler, T args, Action nextCommandHandler, CommandExecutionContext executionContext)
   at Microsoft.VisualStudio.UI.Text.Commanding.Implementation.EditorCommandHandlerService.<>c__DisplayClass13_1`1.<Execute>b__1()
   at Microsoft.VisualStudio.Text.Utilities.GuardedOperations.CallExtensionPoint(Object errorSource, Action call, Predicate`1 exceptionFilter) --- End of stack trace from previous location where exception was thrown ---
    at Microsoft.VisualStudio.Telemetry.WindowsErrorReporting.WatsonReport.GetClrWatsonExceptionInfo(Exception exceptionObject)
--
```

### PR DESCRIPTION
**Pull request by ryzngard**
Fixes #38525

### CODE REVIEW

**Comment by ryzngard** [comment]
I couldn't figure out an easy way to add a unit test for this as the TestWorkspace overrides the method I'm using. Is there a clean way to add this?

File: src/EditorFeatures/Core.Wpf/InlineRename/AbstractEditorInlineRenameService.SymbolRenameInfo.cs [hunk 1]
```diff
@@ -244,7 +244,9 @@ public bool TryOnAfterGlobalSymbolRenamed(Workspace workspace, IEnumerable<Docum
 
             public InlineRenameFileRenameInfo GetFileRenameInfo()
             {
-                if (RenameSymbol.Kind == SymbolKind.NamedType)
+                if (RenameSymbol.Kind == SymbolKind.NamedType &&
+                    _document.Project.Solution.Workspace.CanApplyChange(ApplyChangesKind.ChangeDocumentInfo))
+
```
Comment (CyrusNajmabadi) [inline src/EditorFeatures/Core.Wpf/InlineRename/AbstractEditorInlineRenameService.SymbolRenameInfo.cs]:
```suggestion
```

**Comment by genlu** [comment]
> I couldn't figure out an easy way to add a unit test for this as the TestWorkspace overrides the method I'm using. Is there a clean way to add this?

Depends on what has failed the `CheckAllowedProjectChanges`, you might be able to add a new test workspce XML attribute to do this.

For example:
http://source.roslyn.io/#Roslyn.Services.Test.Utilities/Workspaces/TestWorkspace_XmlConsumption.cs,664

**Comment by ryzngard** [comment]
The root cause is that it's disallowed by the workspace implementation at https://github.com/dotnet/roslyn/blob/master/src/VisualStudio/Core/Def/Implementation/ProjectSystem/MiscellaneousFilesWorkspace.cs#L384 , but there's not much information on _why_. Hopefully this fix is reasonable in that we know we'll be modifying document info if you rename a file, so it's a requirement for the feature to work. Why it may or may not be enabled is another case. I do worry that it doesn't provide a good user experience... 

Adding another test workspace seems reasonable to test. I'll take that route

**Comment by genlu** [comment]
hmm, not sure why a document from shared project is in a misc workspace, it sounds fishy to me...

**Comment by ryzngard** [comment]
I can only report what I was seeing in the debugger :P 

Luckily this doesn't necessarily depend on whatever the behavior is or should be, but rather just more explicitly states that for file rename to work document info needs to be able to be modified.

**Pull request merged**